examples/stencils.cpp

```diff
@@ -0,0 +1,237 @@
+#include <cstdio>
+#include <cstdint>
+#include <chrono>
+#include <thread>
+#include <iostream>
+#include <memory>
+#include <map>
+#include <mutex>
+
+#include <jsapi.h>
+#include <js/CompilationAndEvaluation.h>
+#include <js/Conversions.h>
+#include <js/Initialization.h>
+#include <js/SourceText.h>
+#include <js/experimental/JSStencil.h>
+#include <js/experimental/CompileScript.h>
+
+#include "boilerplate.h"
+
+// This example illustrates how to use Stencils to optimise performance
+// by avoiding repetitive compilations. It shows how to handle
+// compilation errors, otherwise it does no error handling and simply exits
+// if something goes wrong.
+//
+// To reuse Stencils in multiple threads, you must create a JS::FrontendContext
+// in each thread that compiles Javascript.
```

Comment:
Just to be clear, you can pass `JS::FrontendContext` between threads, with the following restriction:
  * `JS::SetNativeStackQuota` must be called in the thread that is going to use the frontend context
  * the frontend context shouldn't be used from 2 threads concurrently

but of course this comment sounds good.

Reply:
I changed the comments.